res/controllers/common-controller-scripts.js

```diff
@@ -392,10 +393,20 @@ bpm.tapButton = function(deck) {
     var now = new Date()/1000;   // Current time in seconds
```

Comment:
Each deck has its own `bpm_tap` control, which has a low pass filter implemented in C++, and the skins use it for tap buttons. Instead of improving this logic, what about deleting it and setting the bpm_tap control here?

Reply:
This bpm tap works different from the mixxxcontrol: This one is far more useful in my opinion as it allows one to roughly match the playback speed of the Deck to an external source. The mixxxcontrol on the other hand just manipulates the beatgrid which is only useful when you have to manually set the bpm of a track (which happens rarely). IMO this is far more useful in most scenarios than the  mixxxcontrol.

Reply:
Gotcha -- we could change the logic of the engine implementation instead? I think it makes most sense if whatever the logic is, a tap button in a skin and a tap button on a controller does the same thing.

Reply:
https://github.com/mixxxdj/mixxx/blob/1c7785847bd7d89866ddc5cc4e759639331f91d3/src/engine/bpmcontrol.cpp#L212-L226

Reply:
The primary reason I wanted fix this was because the feature is broken without [the first commit](https://github.com/mixxxdj/mixxx/pull/1811/commits/7e5dcf37cacb2079b9f878137b9025b972075eb9) and while I was testing, I figured that it might be a good Idea to add some sort of filter to prevent accidental double presses and cases where you just missed the button. The reason why I didn't think about fixing this in the code, is because it had already worked that way in 1.9.0 but was deprecated shortly after in 1.9.2, which is why I figured that there must be a reason why it works this way. IMO, both implementations have use cases but the proposed bpm.tapButton is the far more useful and intuitive version. If I would port the proposed behavior to c++, I would like to keep the old behavior as well (maybe accessible via a separate CO) as it does have its use case when trying to manually determine the bpm of a track.

Reply:
I'm not really understanding the use case for using a BPM tap button to change the playback tempo of a deck. Why doesn't moving the rate slider work? Doesn't the rate slider give you more precise control than attempting to tap the exact tempo of an external source?

Reply:
Yes, but a tap button allows the user to get a rough number for the speed hes working with and it _usually_ also syncs the phase to the taps which is difficult when you are only working with the bpm slider (atleast for beginners and intermediates).